iocore/net/QUICNetVConnection.cc

```diff
@@ -513,14 +513,24 @@ QUICNetVConnection::state_handshake(int event, Event *data)
   switch (event) {
   case QUIC_EVENT_PACKET_READ_READY: {
     QUICPacketCreationResult result;
-    QUICPacketUPtr packet = this->_dequeue_recv_packet(result);
-    if (result == QUICPacketCreationResult::NOT_READY) {
-      error = QUICErrorUPtr(new QUICNoError());
-    } else if (result == QUICPacketCreationResult::FAILED) {
-      error = QUICConnectionErrorUPtr(new QUICConnectionError(QUICTransErrorCode::TLS_FATAL_ALERT_GENERATED));
-    } else {
-      error = this->_state_handshake_process_packet(std::move(packet));
-    }
+    do {
+      QUICPacketUPtr packet = this->_dequeue_recv_packet(result);
+      if (result == QUICPacketCreationResult::NOT_READY) {
+        error = QUICErrorUPtr(new QUICNoError());
+      } else if (result == QUICPacketCreationResult::FAILED) {
+        error = QUICConnectionErrorUPtr(new QUICConnectionError(QUICTransErrorCode::TLS_FATAL_ALERT_GENERATED));
+      } else if (result == QUICPacketCreationResult::SUCCESS) {
+        error = this->_state_handshake_process_packet(std::move(packet));
+      }
+
+      // if we complete handshake, switch to establish state
+      if (this->_handshake_handler && this->_handshake_handler->is_completed()) {
```

Comment:
new change here. we should switch to next state and handle next packet  immediately if we complete handshake.

Reply:
Looks reasonable.